web/src/app/components/LoginComponents/LoginForm.tsx

```diff
@@ -9,25 +9,44 @@ export default function LoginForm() {
     const router = useRouter();
 
     const [inputValue, setInputValue] = useState('');
+    const [inputError, setInputError] = useState(false);
 
 
     const handleEnterGameEvent = () => {
         
         //TODO: add better validation here if required.
         if (inputValue.trim().length != 0) {
-            sessionStorage.setItem('username', inputValue)
-            router.push("1234")
+            sessionStorage.setItem('username', inputValue);
+            router.push("1234");
+        } else {
+            setInputError(true);
         } 
-            //handle error validation in the this clause
+        
+    }
+
+    const handleAsGuest = () => {
+        router.push("1234")
 
+        if (sessionStorage.getItem("username") ) {
+            sessionStorage.removeItem('username')
+        }
     }
 
     useEffect(() => {
+
```

Comment:
i think this is a cleaner way of writing this

```
let username: string = sessionStorage.getItem("username");
if(username) {
   setInputValue(username)
}
```

Reply:
or, I guess it would be 

```
let username: string | undefined = sessionStorage.getItem("username");
if(username) {
   setInputValue(username)
}

```

I like explicitly handling the undefined like this, similar to what I did server.ts